src/main/java/com/alistats/discorki/tasks/CheckJustInGameTask.java

```diff
@@ -36,25 +39,43 @@ public void checkJustInGame() {
         ArrayList<Summoner> skiplist = new ArrayList<Summoner>();
 
         // Get all tracked summoners from the database
-        ArrayList<Summoner> summoners = summonerRepo.findByIsTracked(true).orElseThrow();
-        summoners
+        ArrayList<Summoner> summonersToCheck = summonerRepo.findByTracked(true).orElseThrow();
+
+        // Get all matches in progress from the database
+        ArrayList<Match> matchesInProgress = matchRepo.findByStatus(Status.IN_PROGRESS).orElseThrow();
+
+        // Create a list of summoners that are in a match
+        ArrayList<Summoner> summonersInMatch = new ArrayList<Summoner>();
+        for (Match match : matchesInProgress) {
+            summonersInMatch.addAll(match.getTrackedSummoners());
+        }
+
+        // Define temp game for storing the current game to reduce api calls
+        AtomicReference<CurrentGameInfoDto> tempGame = new AtomicReference<CurrentGameInfoDto>();
+        summonersToCheck
                 .stream()
-                .filter(s -> !s.isInGame())
                 .filter(s -> !skiplist.contains(s))
-                .filter(s -> getCurrentGame(s.getId()) != null)
+                .filter(s -> !summonersInMatch.contains(s))
+                .filter(s -> {
+                    CurrentGameInfoDto currentGameInfoDto = getCurrentGame(s.getId());
+                    if (currentGameInfoDto != null) {
+                        tempGame.set(currentGameInfoDto);
+                        return true;
+                    }
+
+                    return false;
+                })
                 .forEach(s -> {
                     // Get participants from current game and check if other
                     // tracked summoners are in the game. If so, add to skiplist
```

Comment:
This probably wouldnt work, since all summoners have already passed the skiplist filter before any are added.

Reply:
The skiplist contains summoners participating in a match with another checked participant. This is done to reduce API calls (we know they are in a game from earlier retrieved data, so why recheck it).

The filter with the return false statement checks if the summoner is in-game.

So they have two different purposes. Please let me know if I misinterpreted your comment.

Reply:
I think you did misinterpret. However, now you mention the filter at 62-69, I think that do this:
```java
                    currentGameFromDb = // get the latest known game info from match table?
                    if (currentGameInfoDto != currentGameFromDb) {
```
instead of
```java
                    if (currentGameInfoDto != null) {
```

This will trigger a check for the game that has finished, if the player joined a new game in the meantime (instead of skipping it?)

---

Back on the point of the skiplist, its not populated anywhere right?

Reply:
Excellen point. I'll create an issue for this.

The skiplist is populated at the end of the foreach.

Reply:
>The skiplist is populated at the end of the foreach

oh wups, thought it was new with this PR. This looks fine now ;)